libvips/conversion/arrayjoin.c

```diff
@@ -112,16 +114,24 @@ vips_arrayjoin_gen(VipsRegion *out_region,
 	 * forward the request.
 	 */
 	if (width == 1 && height == 1) {
+		VipsRect need;
+
 		i = VIPS_MIN(n - 1, left + top * join->across);
 
-		reg = vips_region_new(in[i]);
+		/* The part of in[i] we need.
+		 */
+		need = out_region->valid;
+		need.left -= join->rects[i].left;
+		need.top -= join->rects[i].top;
 
-		if (vips__insert_just_one(out_region, reg,
```

Comment:
It looks like `vips__insert_just_one()` can be made static after this.
https://github.com/libvips/libvips/blob/3698b934969b00e8d7497186d64f8f18a49b68cf/libvips/conversion/insert.c#L118-L123

Reply:
Ah good point, I'll do that too.